pcdsutils/qt/designer_display.py

```diff
@@ -0,0 +1,67 @@
+"""
+Helper for using designer to layout GUIs.
+"""
+from pathlib import Path
+from typing import ClassVar, Protocol, Type, Union
+
+from qtpy.uic import loadUiType
+
+
+class _UiForm(Protocol):
+    @staticmethod  # <-- this is not entirely true, but PyQt5 uses it as such
+    def setupUi(*args):
+        ...
+
+    @staticmethod
+    def retranslateUi(*args, **kwargs):
+        ...
+
+
+class DesignerDisplay:
+    """Helper class for loading designer .ui files and adding logic."""
+
+    # You must set a filename, the path to your .ui file
+    filename: ClassVar[str]
+    # You can set this for your whole app, override in a subclass, or ignore it
+    ui_dir: ClassVar[Union[str, Path]]
+    # Created automatically
+    ui_form: ClassVar[Type[_UiForm]]
+
+    @classmethod
+    def _load_ui_if_needed(cls):
+        """Load the UI file on first load."""
+        if not hasattr(cls, "ui_form"):
+            try:
+                path = str(Path(cls.ui_dir) / cls.filename)
+            except AttributeError:
+                path = str(cls.filename)
+            cls.ui_form, _ = loadUiType(path)
+
+    def __init__(self, *args, **kwargs):
+        """Apply the file to this widget when the instance is created"""
+        self._load_ui_if_needed()
```

Comment:
in atef we call this with an `__init_subclass__` hook, which seems at first glance to be more or less the same as this?  (loading the ui file if it's not been already?)

I'm fine with it either way, but was curious about the change.

Reply:
I think this is so we don't have to have the ui directory defined at import time for our subdisplays

Reply:
My logic for this route was that on-import `.ui` compiling may slow down applications that don't actually use the displays. In this case, there's only a loading penalty on first usage instead.